File: Bot_Tests/Puppetter/Head-Webdriver-False/friendly_captcha_head_webdriver_false.js
const puppeteer = require('puppeteer');

(async () => {
    const browser = await puppeteer.launch({
        headless: false,
        args: ['--start-maximized'],
    });
    const page = await browser.newPage();
    const { width, height } = await page.evaluate(() => {
        return {
            width: screen.width,
            height: screen.height
        };
    });

    await page.evaluateOnNewDocument(() => {
        Object.defineProperty(navigator, 'webdriver', {
            get: () => undefined
        });
    });

    await page.setViewport({ width, height });
    await page.goto('https://captchatest.online/friendly_captcha');
    await page.type('xpath=//*[@id="username"]', 'John');
    await page.type('xpath=//*[@id="password"]', 'Doe');

    // Click on the checkbox
    await new Promise(resolve => setTimeout(resolve, 3000));

    // Listen for all responses after the click
    page.on('response', async (response) => {
        const request = response.request();
        if (request.method() === 'POST') {
            console.log(`Received response for ${request.url()}`);
            if (response.status() === 200) {
                console.log('Success: POST request was successful');
            } else if (response.status() >= 400) {
                console.log(`Error: POST request failed with status ${response.status()}`);
            }
        }
    });

    await page.click('xpath=/html/body/div[1]/main/form/button');
})();

File: Bot_Tests/Selenium/Head-Webdriver-False/friendly_captcha_head_webdriver_false.py
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.chrome.options import Options
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from time import sleep

options = Options()
options.binary_location = "/usr/bin/chromium"
options.add_experimental_option("detach", True)
service = Service(executable_path="/usr/bin/chromedriver")
driver = webdriver.Chrome(service=service, options=options)

driver.execute_cdp_cmd("Page.addScriptToEvaluateOnNewDocument", {
    "source": """
    Object.defineProperty(navigator, 'webdriver', {
     get: () => undefined
    })
  """
})

driver.get('https://captchatest.online/friendly_captcha')

WebDriverWait(driver, 10).until(
    EC.element_to_be_clickable((By.XPATH, '//*[@id="username"]')))
driver.find_element(By.XPATH, '//*[@id="username"]').send_keys('John')

WebDriverWait(driver, 10).until(
    EC.element_to_be_clickable((By.XPATH, '//*[@id="password"]')))
driver.find_element(By.XPATH, '//*[@id="password"]').send_keys('Doe')

sleep(3)

submit_button_xpath = '/html/body/div[1]/main/form/button'
WebDriverWait(driver, 10).until(EC.element_to_be_clickable((By.XPATH, submit_button_xpath)))
driver.find_element(By.XPATH, submit_button_xpath).click()
